NGitLab.Mock/Clients/LabelClient.cs

```diff
@@ -11,49 +12,109 @@ public LabelClient(ClientContext context)
         {
-            throw new NotImplementedException();
+            var project = FindProject(label.Id) ?? throw new InvalidOperationException($"Cannot find project with ID {label.Id}");
+            var l = FindLabel(project.Labels, label.Name) ?? throw new InvalidOperationException($"Cannot find label '{label.Name}'");
+
+            if (!string.IsNullOrEmpty(label.NewName))
+            {
+                l.Name = label.NewName;
+            }
+
+            if (!string.IsNullOrEmpty(label.Color))
+            {
+                l.Color = label.Color;
+            }
+
+            if (label.Description != null)
+            {
+                l.Description = label.Description;
+            }
+
+            return l.ToClientLabel();
+        }
+
+        public Models.Label EditGroupLabel(LabelEdit label)
+        {
+            var group = FindGroup(label.Id) ?? throw new InvalidOperationException($"Cannot find group with ID {label.Id}");
+            var l = FindLabel(group.Labels, label.Name) ?? throw new InvalidOperationException($"Cannot find label '{label.Name}'");
+
+            if (!string.IsNullOrEmpty(label.NewName))
+            {
+                l.Name = label.NewName;
+            }
+
+            if (!string.IsNullOrEmpty(label.Color))
+            {
+                l.Color = label.Color;
+            }
+
+            if (label.Description != null)
+            {
+                l.Description = label.Description;
+            }
+
+            return l.ToClientLabel();
+        }
+
+        public IEnumerable<Models.Label> ForGroup(int groupId)
+        {
+            var group = FindGroup(groupId) ?? throw new InvalidOperationException($"Cannot find group with ID {groupId}");
+            return group.Labels.Select(x => x.ToClientLabel());
+        }
+
+        public IEnumerable<Models.Label> ForProject(int projectId)
+        {
+            var project = FindProject(projectId) ?? throw new InvalidOperationException($"Cannot find project with ID {projectId}");
+            return project.Labels.Select(x => x.ToClientLabel());
         }
 
-        public Label EditGroupLabel(LabelEdit label)
+        public Models.Label GetGroupLabel(int groupId, string name)
         {
-            throw new NotImplementedException();
+            var group = FindGroup(groupId) ?? throw new InvalidOperationException($"Cannot find group with ID {groupId}");
+            return FindLabel(group.Labels, name)?.ToClientLabel();
         }
 
-        public IEnumerable<Label> ForGroup(int groupId)
+        public Models.Label GetLabel(int projectId, string name)
         {
-            throw new NotImplementedException();
+            var project = FindProject(projectId) ?? throw new InvalidOperationException($"Cannot find project with ID {projectId}");
+            return FindLabel(project.Labels, name)?.ToClientLabel();
         }
 
-        public IEnumerable<Label> ForProject(int projectId)
+        private Project FindProject(int id)
```

Comment:
Not needed, you should be able to use the `GetProject` method defined in the base class